common/test/run-wireaddr.c

```diff
@@ -287,6 +224,31 @@ int main(int argc, char *argv[])
 	expect->u.unresolved.port = 1234;
 	assert(wireaddr_internal_eq(&addr, expect));
 
+	const char raw_input[] =
+		"\x04\x00\x05\xff\x69\x74\x61\x61\x3d\x74\x69\x74\x75\xa0\x74\x69"
+		"\xa0\x74\xa0\xa0\x83\x83\x83\x83\x83\x83\x83\xfe\xff\xff\xff\xff"
+		"\xff\xff\xd7\x83\x83\x83\x83\xa0\xa0\xa0\xa0\xa0\xa0\x74\x69\xa0"
+		"\x74\xa0\xa0\x83\x83\x83\x83\x83\x83\x83\xfe\xff\xff\xff\xff\xff"
+		"\xff\xd7\x83\x83\x83\x83\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0"
+		"\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\x61"
+		"\x75\x74\x69\x74\x61\x69\x61\x75\x74\x61\x75\x72\x3a\x63\x72\x3a"
+		"\x2e\x6f\x6e\x69\x6f\x6e\x74\x74\x69\x75\x74\x74\x69\xa0\xa0\xa0"
+		"\xa0\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83"
+		"\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83\xa0\xa0\xa0\xa0\xa0"
+		"\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\xa0\x61\x75"
+		"\x74\x69\x74\x61\x69\x61\x75\x74\x61\x75\x72\x3a\x63\x72\x69\x74"
+		"\x61\x61\x74\x74\x69\x75\x74\x74\x69\x61\x74\x69\x74\x61\x61\x75"
+		"\x3a\x2e\x6f\x6e\x69\x6f\x6e\x74\x74\x69\x75\x74\x3a\x74\x69\xa0"
+		"\xa0\xa0\xa0\x83\x83\x83\x83\x83\x83\x83\x04\x83\x83\x83\x83\x83"
+		"\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83\x83"
+		"\xa0\x72\x3a\x26\x07";
+
+	const u8 *crashing_input = tal_dup_arr(tmpctx, u8, (const u8*) raw_input, sizeof(raw_input) - 1, 0);
+	size_t crashing_input_len = tal_bytelen(crashing_input);
+
+	struct wireaddr_internal decoded_wa;
+	assert(fromwire_wireaddr_internal(&crashing_input, &crashing_input_len, &decoded_wa));
```

Comment:
Can we improve the test to check the expected decoded values, as with the above tests?